File: Content/example-code/HelloWorldEnvironment/csharp/HelloWorldEnvironment.cs
//version 1.0.1
namespace Demos
{
    using System;
    using System.Drawing;
    using Applitools.Selenium;
    using OpenQA.Selenium;
    using OpenQA.Selenium.Chrome;

    public class HelloWorldSimple
    {
        public static void Main(string[] args)
        {
            // Initialize the eyes SDK and set your private API key.     
            var serverURL = new Uri("https://eyesapi.applitools.com");
            // %%%% start set-baseline-env
            var eyes = new Eyes(serverURL);
            // %%%% stop set-baseline-env
           
            
            eyes.ApiKey = Environment.GetEnvironmentVariable("APPLITOOLS_API_KEY");

            // Open a Chrome browser.
            var innerDriver = new ChromeDriver();
            var viewportSize = new Size(1024, 768);
            // Start the test and set the browser's viewport size
            eyes.SetAppEnvironment("Windows 10" ,"Firefox");
            // %%%% start set-baseline-env
            eyes.BaselineEnvName = "desktop browser";
            //now call open
            // %%%% stop set-baseline-env
            // %%%% start set-matchlevel
            eyes.MatchLevel = Applitools.MatchLevel.Layout;
            // %%%% stop set-matchlevel
            var driver = eyes.Open(innerDriver, "cross environment", "test 1", viewportSize);
            // %%%% stop set-baseline-env
            try
            {		
                // Navigate the browser to the "hello world!" web-site.
                string website = "https://applitools.com/helloworld";
                driver.Url = website;

                // Visual checkpoint #1
                eyes.CheckWindow("Before mouse click");

                // Click the "Click me!" button.
                driver.FindElement(By.TagName("button")).Click();

                // Visual checkpoint #2.
                eyes.CheckWindow("After mouse click");

                // End the test
                var throwtTestCompleteException = false;
                Applitools.TestResults result = eyes.Close(throwtTestCompleteException); 
                string url = result.Url;
                if (result.IsNew)
                {
                    Console.WriteLine("New Baseline Created: URL=" + url);
                }
                else if (result.IsPassed)
                {
                    Console.WriteLine("All steps passed:     URL=" + url);
                }
                else
                {
                    Console.WriteLine("Test Failed:          URL=" + url);
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex);
            }
            finally
            {
                // If the test was aborted before eyes.Close was called, ends the test as aborted.
                eyes.AbortIfNotClosed();
            }
            // Close the browser.
            innerDriver.Quit();
        }
    }
}

File: Content/example-code/HelloWorldEnvironment/javascript/HelloWorldEnvironment.js
//version 1.0.2

    var Eyes = require('eyes.selenium').Eyes;
    var webdriver = require('selenium-webdriver');
// %%%% start set-matchlevel
    var MatchLevel = require('eyes.selenium').MatchLevel;
    //....
// %%%% stop set-matchlevel
    var innerDriver;
    var eyes;

function main() {

// Initialize the eyes SDK and set your private API key.

    var serverURL = "https://eyesapi.applitools.com";
   
   // %%%% start set-baseline-env
    eyes = new Eyes(serverURL); 
    // %%%% stop set-baseline-env
    
    var apiKey = process.env.APPLITOOLS_API_KEY;
    eyes.setApiKey(apiKey);


    // Open a Chrome browser.
    innerDriver = new webdriver.Builder()
        .withCapabilities(webdriver.Capabilities.chrome())
        .build();

    // Start the test and set the browser's viewport size to 1024 x 768.

    var viewportSize = {width: 1024, height: 768};

    // %%%% start set-matchlevel  
    eyes.setMatchLevel(MatchLevel.Layout);
    // %%%% stop set-matchlevel

    // %%%% start set-baseline-env
    eyes.setBaselineEnvName("desktop browser");
    //now call open....
    // %%%% stop set-baseline-env
    eyes.setHostingApp("Firefox");
    eyes.setHostOS("Windows 7");
    eyes.open(innerDriver, 'cross environment', 'test 1', viewportSize)
        .then(function (driver){ afterOpen(driver)});
   
}

function afterOpen(driver) {
    try {
        // insert your checkpoints here....

        // Navigate the browser to the "hello world!" web-site.
        var  website = "https://applitools.com/helloworld";
        driver.get(website);

        eyes.checkWindow('Before mouse click');  // Visual checkpoint #1.

        // Click the "Click me!" button.
        driver.findElement(webdriver.By.tagName('button')).click();


        eyes.checkWindow('After mouse click'); // Visual checkpoint #2.


        // End the test.

        var throwtTestCompleteException = false;
        eyes.close(throwtTestCompleteException)
        .then(function(result) {

            var url = result.appUrls.session;
            if (result.isNew) {
                console.log("New Baseline Created: URL=" + url);
            } else if (result.isPassed) {
                console.log("All steps passed:     URL=" + url);
            } else {
                console.log("Test Failed:          URL=" + url);
                }            
        });


    } finally {
      // If the test was aborted before eyes.close was called ends the test as aborted.
      eyes.abortIfNotClosed(); //could add .then if necessary
    }

    if (innerDriver) {
        innerDriver.quit(); // Close the browser.
    } 
}

main();
